realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -563,6 +563,12 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
                 "Realm", "realm", className, "object", "boolean", "update", "Map<RealmObject,RealmObjectProxy>", "cache" // Argument type & argument name
         );
 
+        writer
+            .beginControlFlow("if (((RealmObject) object).realm != null && ((RealmObject) object).realm.threadId != realm.threadId)")
+                .emitStatement("throw new IllegalArgumentException(\"Objects belong to another Realm instance in other" +
```

Comment:
Perhaps "Objects belonging to Realm instances in other threads cannot be copied into this Realm instance"